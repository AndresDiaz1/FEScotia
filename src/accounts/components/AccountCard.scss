.account-card {
    &__container {
        border: 2px solid black;
        background-color: white;
    }
    &__link {
        text-decoration: underline;
        cursor: pointer;
    }
}