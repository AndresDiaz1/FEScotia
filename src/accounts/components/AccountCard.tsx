import React from 'react';
import { IonGrid, IonLabel, IonRow } from '@ionic/react';
import './AccountCard.scss';
import { Account } from '../AccountTypes';
import { formatCurrency } from '../../utils/Utils';
import { PATHS } from '../../AppConstants';
import { useHistory } from 'react-router';
import { useDispatch } from 'react-redux';
import { fetchDetails } from '../../details/DetailsStore';

const AccountCard: React.FC<Account> = ({ id, accountType, availableValue }) => {
  const history = useHistory();
  const dispatch = useDispatch();

  const goToDetail = () => {
    dispatch(fetchDetails(parseInt(id)));
    history.push(PATHS.DETAILS);
  };

  return (
    <div className="account-card__container ion-padding">
      <IonGrid>
        <IonRow>
          <IonLabel> &middot; {id} </IonLabel>
          <IonLabel> - {accountType}</IonLabel>
        </IonRow>
        <IonRow>
          <IonLabel>{formatCurrency(availableValue)}</IonLabel>
        </IonRow>
        <IonRow>
          <IonLabel>Available amount</IonLabel>
        </IonRow>
        <IonRow className="ion-justify-content-center">
          <IonLabel color="primary" className="account-card__link" onClick={() => goToDetail()}>
            See Details
          </IonLabel>
        </IonRow>
      </IonGrid>
    </div>
  );
};

export default AccountCard;
