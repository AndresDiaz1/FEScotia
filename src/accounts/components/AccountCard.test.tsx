import React from 'react';
import AccountCard from './AccountCard';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

describe('AccountCard component', () => {
  describe('Initial state', () => {
    it('Should render', () => {
      const account = {
        id: '1',
        clientId: 1,
        accountType: 'savings',
        status: 60,
        availableValue: 33,
      };

      const state = {
        accounts: {
          isLoading: false,
          accounts: [account],
          errorMessage: '',
        },
      };
      const store = configureStore([thunk])(state);

      const component = mount(
        <Provider store={store}>
          <AccountCard {...account} />
        </Provider>
      );
      expect(component).toMatchSnapshot();
    });
  });
});
