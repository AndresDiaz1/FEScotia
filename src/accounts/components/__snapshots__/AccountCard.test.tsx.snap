// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountCard component Initial state Should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AccountCard
    accountType="savings"
    availableValue={33}
    clientId={1}
    id="1"
    status={60}
  >
    <div
      className="account-card__container ion-padding"
    >
      <ForwardRef(IonGrid)>
        <IonGrid
          forwardedRef={null}
        >
          <ion-grid>
            <ForwardRef(IonRow)>
              <IonRow
                forwardedRef={null}
              >
                <ion-row>
                  <ForwardRef(IonLabel)>
                    <IonLabel
                      forwardedRef={null}
                    >
                      <ion-label>
                         · 
                        1
                         
                      </ion-label>
                    </IonLabel>
                  </ForwardRef(IonLabel)>
                  <ForwardRef(IonLabel)>
                    <IonLabel
                      forwardedRef={null}
                    >
                      <ion-label>
                         - 
                        savings
                      </ion-label>
                    </IonLabel>
                  </ForwardRef(IonLabel)>
                </ion-row>
              </IonRow>
            </ForwardRef(IonRow)>
            <ForwardRef(IonRow)>
              <IonRow
                forwardedRef={null}
              >
                <ion-row>
                  <ForwardRef(IonLabel)>
                    <IonLabel
                      forwardedRef={null}
                    >
                      <ion-label>
                        COP 33.00
                      </ion-label>
                    </IonLabel>
                  </ForwardRef(IonLabel)>
                </ion-row>
              </IonRow>
            </ForwardRef(IonRow)>
            <ForwardRef(IonRow)>
              <IonRow
                forwardedRef={null}
              >
                <ion-row>
                  <ForwardRef(IonLabel)>
                    <IonLabel
                      forwardedRef={null}
                    >
                      <ion-label>
                        Available amount
                      </ion-label>
                    </IonLabel>
                  </ForwardRef(IonLabel)>
                </ion-row>
              </IonRow>
            </ForwardRef(IonRow)>
            <ForwardRef(IonRow)
              className="ion-justify-content-center"
            >
              <IonRow
                className="ion-justify-content-center"
                forwardedRef={null}
              >
                <ion-row>
                  <ForwardRef(IonLabel)
                    className="account-card__link"
                    color="primary"
                    onClick={[Function]}
                  >
                    <IonLabel
                      className="account-card__link"
                      color="primary"
                      forwardedRef={null}
                      onClick={[Function]}
                    >
                      <ion-label
                        color="primary"
                        onClick={[Function]}
                      >
                        See Details
                      </ion-label>
                    </IonLabel>
                  </ForwardRef(IonLabel)>
                </ion-row>
              </IonRow>
            </ForwardRef(IonRow)>
          </ion-grid>
        </IonGrid>
      </ForwardRef(IonGrid)>
    </div>
  </AccountCard>
</Provider>
`;
